bootstrap/SubjectRepositoryBuilder.go

```diff
@@ -0,0 +1,58 @@
+package bootstrap
+
+import (
+	"authz/bootstrap/serviceconfig"
+	"authz/domain/contracts"
+	"authz/infrastructure/repository/userservice"
+	"crypto/tls"
+	"crypto/x509"
+	"net/http"
+	"net/url"
+)
+
+type SubjectRepositoryBuilder struct {
+	config *serviceconfig.ServiceConfig
+}
+
+func NewSubjectRepositoryBuilder() *SubjectRepositoryBuilder {
+	return &SubjectRepositoryBuilder{}
+}
+
+func (s *SubjectRepositoryBuilder) WithConfig(config *serviceconfig.ServiceConfig) *SubjectRepositoryBuilder {
+	s.config = config
+	return s
+}
+
+func (s *SubjectRepositoryBuilder) Build() (contracts.SubjectRepository, error) {
+	config := s.config.UserServiceConfig
+
+	return createUserServiceSubjectRepository(config)
+}
+
+func createUserServiceSubjectRepository(config serviceconfig.UserServiceConfig) (contracts.SubjectRepository, error) {
+	url, err := url.Parse(config.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	caCert, err := x509.SystemCertPool()
```

Comment:
No, but FWIW, I think we should probably make it work. :) I think we should handle CA certs at the OS level.

Reply:
Adding it to the TODO doc.

Reply:
Probably this will work as we have used the WithSystemCerts() method in our spiceDB connection, which uses the same underlying method as done here, but we can verify the same in our downstream deployment later